expression/util.go

```diff
@@ -191,7 +191,7 @@ func SubstituteCorCol2Constant(expr Expression) (Expression, error) {
 	case *Constant:
 		if x.DeferredExpr != nil {
 			newExpr := FoldConstant(x)
-			return &Constant{Value: newExpr.(*Constant).Value, RetType: x.GetType()}, nil
+			return &Constant{Value: newExpr.(*Constant).Value, RetType: x.GetType(), DeferredExpr: x.DeferredExpr}, nil
```

Comment:
Since we rely on `DeferredExpr` to decide whether this constant is stable, we should maintain `DeferredExpr` properly as we do in `foldConstant`. Hence, in this function, we should handle `CorrelatedColumn` and `ScalarFunction` branch as well IMHO.

Reply:
@eurekaka, Thank you for the comment. I agree that the PR should also handle ScalarFunction like as done in Constant branch except for CorrelatedColumn.  I've update the PR.

BTW, if CorrelatedColumn can be parameterized like Constant, its struct should be extended to hold DeferredExpr like Constant. Otherwise, we can leave it.

Reply:
Thank you for the correction. I've reverted the changes.